homework_1/task2/progressivetax/progressiveTax.go

```diff
@@ -0,0 +1,71 @@
+package progressivetax
+
+import (
+	"github.com/pkg/errors"
+)
+
+type Bracket struct {
+	TaxRate   float32
+	Threshold float32
+}
+
+func ValidateIncomeAndTaxBrackets(income float32, taxBrackets []Bracket) error {
+	if income < 0 {
+		return errors.New("income is lower than 0.0")
+	}
+	if taxBrackets[0].Threshold != 0.0 {
+		return errors.New("threshold of the first bracket is not 0.0")
+	}
+
+	if len(taxBrackets) < 2 {
+		return errors.New("there is less than 2 tax brackets")
+	}
```

Comment:
Why would we require at least 2 tax brackets? It is perfectly fine to have a single tax bracket - e.g. a tax of 20% is applied on the income.

Reply:
I've also thought of that, but after making my file, I checked PR's from other cadets and saw this check. It was not mentioned as wrong so I've also added it to mine.
I've removed it now.